src/mumble/widgets/TrayIcon.h

```diff
@@ -0,0 +1,46 @@
+// Copyright The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef MUMBLE_MUMBLE_WIDGETS_TRAYICON_H_
+#define MUMBLE_MUMBLE_WIDGETS_TRAYICON_H_
+
+#include <functional>
+
+#include <QAction>
+#include <QTimer>
+#include <QtWidgets/QMenu>
+#include <QtWidgets/QSystemTrayIcon>
+
+class TrayIcon : public QSystemTrayIcon {
+	Q_OBJECT
+
+public:
+	TrayIcon();
+
+public slots:
+	void on_hideAction_triggered();
+	void on_showAction_triggered();
+	void on_toggleShowHide();
+
+	void on_icon_update();
+	void on_tray_unhighlight();
+
+private:
+	std::reference_wrapper< QIcon > m_statusIcon;
+	bool m_blinkingIcon      = false;
```

Comment:
Thinking about this a bit more, I think an `enum` should make the code even more readable than a `bool`:
```suggestion
private:
	enum class BlinkState {
		RegularIcon,
		BlinkIcon,
	};

	std::reference_wrapper< QIcon > m_statusIcon;
	BlinkState m_blinkState = BlinkState::RegularIcon;
```

Reply:
Sure! Done!